### PR TITLE
fix: correctly handle kernel parameters

### PR DESCRIPTION
**Pull request by haraldh**
The kernel has an odd way to handle `"` surrounded parameters.
To handle the parameters as the kernel would do, no simple shell script
suffices, so a new utility `dracut-util` is introduced. Written in "C"
it handles `dracut-getarg` and `dracut-getargs` as the old shell script
functions `_dogetarg` and `_dogetargs` would.

### CODE REVIEW

File: test/TEST-99-GETARG/test.sh [hunk 1]
```diff
@@ -0,0 +1,94 @@
+#!/bin/bash
+
```
Comment (johannbg) [inline test/TEST-99-GETARG/test.sh:2]: Minor nitpick add proper header for new files ( even if it is an test ) it's missing
```
# This file is part of dracut.
# SPDX-License-Identifier: GPL-2.0-or-later
```

File: util/util.c [hunk 2]
```diff
@@ -0,0 +1,317 @@
+// SPDX-License-Identifier: GPL-2.0
+
```
Comment (johannbg) [inline util/util.c:2]: Should this be GPL-2.0-or-later so a full header would look like this  
```
// This file is part of dracut.
// SPDX-License-Identifier: GPL-2.0-or-later
```
or is the C code GPL-2.0 only?

**Comment by nl6720** [comment]
`make install` doesn't install `dracut-util`.

File: util/util.c (hunk 2 above)
Comment (haraldh) [reply util/util.c:2]: yeah, the argument parsing code is copied from the kernel

File: test/TEST-99-GETARG/test.sh (hunk 1 above)
Comment (haraldh) [reply test/TEST-99-GETARG/test.sh]: will add

**Comment by haraldh** [comment]
> `make install` doesn't install `dracut-util`.

@nl6720 right.. fixed

File: test/TEST-98-GETARG/test.sh [hunk 3]
```diff
@@ -0,0 +1,94 @@
+#!/bin/bash
+
```
Comment (johannbg) [inline test/TEST-98-GETARG/test.sh:2]: Missed.
```
# This file is part of dracut.
# SPDX-License-Identifier: GPL-2.0-or-later
```

Comment (haraldh) [reply test/TEST-98-GETARG/test.sh:2]: done.. this time for real :)

**Review by johannbg — APPROVED** [review]

**Review by danimo — APPROVED** [review]

**Pull request merged**